pkg/cloud/cloud_test.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2022 The Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/fsx"
+	"github.com/golang/mock/gomock"
+	"sigs.k8s.io/aws-file-cache-csi-driver/pkg/cloud/mocks"
+	"testing"
+)
+
+func TestCreateFileCache(t *testing.T) {
+	var (
+		volumeName                                 = "volumeName"
+		mountname                                  = "mountName"
+		fileCacheId                                = "fc-1234"
+		volumeSizeGiB                        int64 = 1200
+		subnetId                                   = "subnet-056da83524edbe641"
+		securityGroupIds                           = []string{"sg-086f61ea73388fb6b", "sg-0145e55e976000c9e"}
+		dnsname                                    = "test.us-east-1.fsx.amazonaws.com"
+		dataRepositoryAssociations                 = "FileCachePath=/ns1/,DataRepositoryPath=nfs://10.0.92.69/,NFS={Version=NFS3},DataRepositorySubdirectories=[subdir1,subdir2,subdir3]"
+		fileCacheType                              = "LUSTRE"
+		fileCacheTypeVersion                       = "2.12"
+		weeklyMaintenanceStartTime                 = "7:00:00"
+		LustreConfiguration                        = []string{"DeploymentType=CACHE_1", "PerUnitStorageThroughput=1000", "MetadataConfiguration={StorageCapacity=2400}"}
+		copyTagsToDataRepositoryAssociations       = true
+		kmsKeyId                                   = "arn:aws:kms:us-east-1:215474938041:key/48313a27-7d88-4b51-98a4-fdf5bc80dbbe"
+		extraTags                                  = []string{"key1=value1", "key2=value2"}
+	)
+	testCases := []struct {
+		name     string
+		testFunc func(t *testing.T)
+	}{
+		{
+			//TODO: Expand test checks after adding more values in FileCache object
```

Comment:
We should add the following tests:
- create w/ an S3 DRA
- create w/ an S3 DRA but there's a invalid path so the create fails
- create w/ NFS but it's missing DNS server IP addresses or Cache path
- CreateFileCacheWithContext return error

Reply:
Since this mocks the API call, I don't believe we can have these tests without adding additional checks in the functions themselves. I think we'd have to leave these for e2e testing. `CreateFileCacheWithContext return error` is also included in this PR